cmd/kaf/completion.go

```diff
@@ -0,0 +1,182 @@
+package main
+
+import (
+	"io"
+	"os"
+	"text/template"
+
+	"github.com/spf13/cobra"
+)
+
+func init() {
+	rootCmd.AddCommand(completionCmd)
+}
+
+var completionCmd = &cobra.Command{
+	Use:       "completion [SHELL]",
+	Short:     "Generate bash completion script for bash or zsh",
+	Args:      cobra.ExactValidArgs(1),
+	ValidArgs: []string{"bash", "zsh"},
+	Run: func(cmd *cobra.Command, args []string) {
+		switch args[0] {
+		case "bash":
+			rootCmd.GenBashCompletion(os.Stdout)
+		case "zsh":
+			if err := genZshCompletion(os.Stdout); err != nil {
+				panic(err)
+			}
+		}
+	},
+}
+
+// GenZshCompletion generates a zsh completion and writes to the passed writer.
```

Comment:
Updated.

Sad, but seems like zsh autocompletion doesn't work in cobra for now. I used this hack to make it work.